packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1980,9 +1991,17 @@ export class StripeHelper {
       if (!this.getCustomerPaypalAgreement(customer)) {
         billingDetails.paypal_payment_error =
           PAYPAL_PAYMENT_ERROR_MISSING_AGREEMENT;
-      } else if (await this.hasOpenInvoice(customer)) {
-        billingDetails.paypal_payment_error =
-          PAYPAL_PAYMENT_ERROR_FUNDING_SOURCE;
+      } else {
+        const invoices = await this.getLatestInvoicesForActiveSubscriptions(
+          customer
+        );
+        if (
+          this.hasOpenInvoice(invoices) &&
```

Comment:
We don't need this half of the predicate.  `some` will return false if the array is empty.

Reply:
If I remove this predicate, the logic will incorrectly return false if there are paid invoices on any active subscriptions, since paymentAttempts for a paid PayPal invoice would also be > 0 (just tried it locally).

@chenba am I misunderstanding you?

Reply:
Oh sorry, I was focusing too much on the addition (green) part of the diff.  You're correct, we need this because of the 'open' status filtering part.  🤦

Reply:
No worries.

Actually after looking at this logic more closely, I think I need to ensure that the _same_ invoice is both open _and_ has paymentAttempts > 0. Otherwise it's possible that the user has e.g. two active subscriptions, one with a paid invoice (and 1 payment attempt) and the other upgraded subscription with an open invoice and 0 payment attempts. Right now they would incorrectly get this error still (though this is a different scenario than the bug filed by QA).